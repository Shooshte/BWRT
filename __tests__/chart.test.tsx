import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { server } from "../src/tests/mock/api/server";
import { http, HttpResponse } from "msw";

import Chart from "../src/app/components/chart";
import { Wrapper } from "../src/tests/test-utils";

describe("Chart", () => {
  it("renders Error message when request fails", async () => {
    server.use(
      http.get("/ping", async () => {
        return new HttpResponse("Internal server error", { status: 500 });
      })
    );

    render(<Chart />, { wrapper: Wrapper });
    const errorMessage = await screen.findByText("Unable to load coin data.");
    expect(errorMessage).toBeInTheDocument();
  });
});
