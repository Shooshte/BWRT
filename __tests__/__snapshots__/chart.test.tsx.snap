// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders chart 1`] = `
<div
  style="height: 100%; width: 100%;"
>
  <div
    class="recharts-responsive-container"
    style="width: 100%; height: 100%; min-width: 0;"
  />
</div>
`;
